### PR TITLE
Detect Regex lib with cmake instead of pkg_config

### PR DESCRIPTION
**Pull request by piponazo**
Apply patch provided in #466 by @cgilles . 

@cgilles : I tried to give you the authorship in the commit but I do not know how to do it properly without your email.

### CODE REVIEW

**Review by clanmills — APPROVED** [review]
This is fine. Thanks very much to @cgilles and @piponazo 

Is it possible to add this change to the RC1 branch, or should I update RC1 manually?

**Comment by piponazo** [comment]
In my opinion, it is better to merge small branches into master step by step, so that we can easily identify if something is wrong in that branch thanks to the CI jobs.

Once this is merged, I can take care of rebasing the **RC1** branch in top of master for you.

File: cmake/FindRegex.cmake [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# - Try to find the Regex library
+#
+# Once done this will define
+#
+#  REGEX_FOUND       - system has libregex
+#  REGEX_INCLUDE_DIR - the libregex include directory
+#  REGEX_LIBRARIES   - Link these to use libregex
+#
+# Copyright (c) 2018, Gilles Caulier, <caulier dot gilles at gmail dot com>
```
Comment (D4N) [inline cmake/FindRegex.cmake]: @piponazo His email is here

Comment (piponazo) [reply cmake/FindRegex.cmake]: Thanks! Situation fixed ;)

**Comment by clanmills** [comment]
I agree with your approach.

I've fixed #468 in RC1 and asked Gilles to use RC1 for his builds.  I'm investigating #465 at the moment.  

Perhaps the simplest thing is for you to incorporate my fix for #468 into 'master'.  When I have a fix for #468, I'll update that with a patch.

I'm lost in the git nightmare again.  I'll go out for a run!

**Pull request merged**